interfaces/builtin/opencl.go

```diff
@@ -0,0 +1,53 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/ubuntu-core/snappy/interfaces"
+)
+
+const openclConnectedPlugAppArmor = `
+# Description: Can access opencl. 
+# Usage: reserved
+
+  # OpenCL Installable Client Driver (ICD) Loader
+  /etc/OpenCL/vendors/** r,
```

Comment:
It fails. But, please, try to see my point (I'm reading your comments trying to find what I did wrong as I pointed out in my first message).

```
$ cat /var/lib/snapd/apparmor/profiles/snap.john-the-ripper.opencl | grep vendors
  /etc/OpenCL/vendors/** r,

$ john-the-ripper.opencl --list=opencl-devices
[ERROR]

$ dmesg
[28834.360820] audit: type=1400 audit(1464031699.616:33): apparmor="DENIED" operation="open" profile="snap.john-the-ripper.opencl" name="/etc/OpenCL/vendors/" pid=7880 comm="john" requested_mask="r" denied_mask="r" fsuid=1002 ouid=0
```

I already tried to be more permissive to be able to load the real library. But the error message is puzzling me.